### PR TITLE
Bug 1910160: Create OperatorConditions for CSVs being installed

### PR DESCRIPTION
**Pull request by awgreene**
Problem: If OLM creates an OperatorConditions for an object before the
CSV is being installed it could update the existing deployment which is
not upgradeable, killing the pod.

Solution: Do not create the OperatorCondition until the CSV is being
installed.

### CODE REVIEW

File: pkg/controller/operators/operatorcondition_controller.go [hunk 1]
```diff
@@ -201,6 +202,13 @@ func (r *OperatorConditionReconciler) ensureDeploymentEnvVars(operatorCondition
 		if err != nil {
 			return err
 		}
+
+		// Check the deployment is owned by a CSV with the same name as the OperatorCondition.
+		deploymentOwner := ownerutil.GetOwnerByKind(deployment, operatorsv1alpha1.ClusterServiceVersionKind)
```
Comment (dinhxuanvu) [inline pkg/controller/operators/operatorcondition_controller.go:207]: Would it be easier to use `IsOwnedBy` instead of this whole if condition?

**Review by dinhxuanvu — CHANGES_REQUESTED** [review]
Looking good. Just a nit comment.

File: pkg/controller/operators/operatorcondition_controller.go (hunk 1 above)
Comment (awgreene) [reply pkg/controller/operators/operatorcondition_controller.go:207]: I would rather not have to make a get call against the api server to check this.

Comment (dinhxuanvu) [reply pkg/controller/operators/operatorcondition_controller.go:207]: Fair enough.

**Review by dinhxuanvu — APPROVED** [review]
/lgtm

**Pull request merged**